src/linear_algebra/GLA.jl

```diff
@@ -156,6 +161,56 @@ plus_lm(x::AbstractVector) = begin
 end
 zero_lm(n::Int) = x::AbstractVector -> zeros(eltype(x), n)
 
+# LinearOps instance
+#-------------------
+
+@auto_hash_equals struct LinearOpDom
+  N::Int
+end
+
+@instance LinearFunctions(LinearOpDom, LinearOperator) begin
+  @import adjoint, +
+
+  dom(f::LinearOperator) = LinearOpDom(size(f,2))
+  codom(f::LinearOperator) = LinearOpDom(size(f,1))
+
+  compose(f::LinearOperator, g::LinearOperator) = g*f
+  id(V::LinearOpDom) = opEye(V.N)
+
+  oplus(V::LinearOpDom, W::LinearOpDom) = LinearOpDom(V.N + W.N)
+  oplus(f::LinearOperator, g::LinearOperator) = oplus_lo(f, g)
+  mzero(::Type{LinearOpDom}) = LinearOpDom(0)
+  braid(V::LinearOpDom, W::LinearOpDom) = braid_lo(LinearOperator, V.N, W.N)
+  mcopy(V::LinearOpDom) =
+    opExtension(1:V.N, 2*V.N)+opExtension((V.N+1):(2*V.N), 2*V.N)
+  delete(V::LinearOpDom) = opZeros(0, V.N)
+  plus(V::LinearOpDom) =
+    opRestriction(1:V.N, 2*V.N)+opRestriction((V.N+1):(2*V.N), 2*V.N)
+  zero(V::LinearOpDom) = opZeros(V.N, 0)
+
+  plus(f::LinearOperator, g::LinearOperator) = f+g
+  scalar(V::LinearOpDom, c::Number) = opEye(typeof(c),V.N)*c
+  antipode(V::LinearOpDom) = scalar(V,-1)
+end
+
+oplus_lo(f::LinearOperator, g::LinearOperator) = begin
+  dom_total   = size(f,2) + size(g,2)
+  codom_total = size(f,1) + size(g,1)
+  dom_f   = 1:size(f,2)
+  codom_f = 1:size(f,1)
+  dom_g   = (size(f,2)+1):dom_total
+  codom_g = (size(f,1)+1):codom_total
+  fOp = opExtension(codom_f, codom_total)*f*opRestriction(dom_f,dom_total)
+  gOp = opExtension(codom_g, codom_total)*g*opRestriction(dom_g,dom_total)
+  fOp + gOp
+end
+braid_lo(::Type{LinearOperator}, v::Int, w::Int) = begin
```

Comment:
Likewise, this can be moved into the body of `@instance` with signature `braid(V::LinearOpDom, W::LinearOpDom)`.

Reply:
Just saw your comment above @jpfairbanks. What is the rationale for `braid(::Type{LinearOperatior}, ...)` instead of a dispatch on object type as in `braid(V::LinearOpDom, W::LinearOpDom)`?

Reply:
The definition of `braid_lo` is meant to just be a helper function. Since `braid(V::LinearOpDom, W::LinearOpDom)` is already present in the body of `@instance`, I think @jpfairbanks was trying to keep from defining two functions with the same signature (he was referring to the helper function, not the in-instance function).
You're right, though, that a helper function can be avoided here and the function can be moved into the body of `@instance`, so I'll make that edit!

Reply:
Yeah I was just explaining the Julia idiom.  I think you should just eliminate that helper function entirely and inline it into the instance definition.